File: BigStudentsDiary/BigStudentsDiary.Domain/Models/Note.cs
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace BigStudentsDiary.Domain.Models;


public class Note
{
    [JsonConstructor]
    private Note(
        Guid noteId,
        Guid studentId,
        int disciplineId,
        int lessonNumber,
        string content,
        DateTime createdAt,
        DateTime updatedAt)
    {
        NoteId = noteId;
        StudentId = studentId;
        DisciplineId = disciplineId;
        LessonNumber = lessonNumber;
        Content = content;
        CreatedAt = createdAt;
        UpdatedAt = updatedAt;
    }
    [Column("NoteID")]
    public Guid NoteId { get; set; }
    
    [Column("StudentID")]
    public Guid StudentId { get; set; }
    
    [Column("DisciplineID")]
    public int DisciplineId { get; set; }
    public int LessonNumber { get; set; }
    public string Content { get; set; } = "";
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public static Note Create(
        Guid noteId,
        Guid studentId,
        int disciplineId,
        int lessonNumber,
        string content,
        DateTime createdAt,
        DateTime updatedAt)
    {
        return new Note(
            noteId,
            studentId,
            disciplineId,
            lessonNumber,
            content,
            createdAt,
            updatedAt);
    }
}

File: TimeTableApp/lib/models/Note.dart
class Note {
  final String noteId;
  final int lessonNumber;
  String content;

  Note({
    required this.noteId,
    required this.lessonNumber,
    required this.content,
  });

  factory Note.fromJson(Map<String, dynamic> json) {
    return Note(
      noteId: json['noteId'],
      lessonNumber: json['lessonNumber'],
      content: json['content'],
    );
  }
  Note copyWith({
    String? noteId,
    int? lessonNumber,
    String? content,
  }) {
    return Note(
      noteId: noteId ?? this.noteId,
      lessonNumber: lessonNumber ?? this.lessonNumber,
      content: content ?? this.content,
    );
  }
  Note.empty(int lessonNumber)
      : noteId = '',
        lessonNumber = lessonNumber,
        content = '';
}